### ISSUE TITLE
Right to Left languages

### ISSUE THREAD

**Issue by medicamin** [jangernert/FeedReader#197]
Hi there,
Nice app. simple and elegant. 
Is that possible to add RTL support for languages such as Persian or Arabic?
all RTL items sre in Left direction

**Comment by jangernert** [comment]
No idea how this would work. No way to test this since I only speak English and German.
So if someone steps in and implements support for it, I would welcome it. But it's nothing I can implement myself.

**Comment by ZanderBrown** [comment]
I'm assuming you have Gtk in RTL mode?

**Comment by medicamin** [comment]
> I'm assuming you have Gtk in RTL mode?

In most of GTK apps which has writing function, if I type English, the sentence begins from the left. and if I type Persian, the sentence begin from the right. for example in gedit,  medit, notes and...
I don't make any change in the app to write from the right. They understand the language! 
look at this picture: https://i.imgur.com/8yoaI6K.png

**Comment by jangernert** [comment]
Like I said: feel free to submit patches, but don't expect me to figure out how to make languages I don't even speak work in FeedReader ;)

**Comment by medicamin** [comment]
Yes I understand. I just wanted to write about that, so if someone could
make it possible.
Thank you anyway.

**Comment by bil-elmoussaoui** [comment]
I will take care!

**Comment by jangernert** [comment]
Any progress?

**Comment by bil-elmoussaoui** [comment]
I will get a new laptop in two/three weeks. I will work on this just after
that :)

Le 7 sept. 2016 00:04, "Jan Lukas Gernert" <notifications@github.com> a
écrit :

**Comment by jangernert** [comment]
Any news on this?

**Comment by bil-elmoussaoui** [comment]
I don't think that i will get the time to make this lands on 2.0 beta, maybe on stable version ;) i was so busy with studies and finals are on two weeks..

**Comment by hadess** [comment]
The interface needs to look mirrored, is pretty much the only requirement. For example, if a widget is placed on the right, and is still on the right when you switch languages, then you have a bug. Ditto if you have misaligned widgets, etc. It usually happens only for custom widgets, or layouts.


You could test with:
```
LC_ALL=he_IL.UTF-8 feedreader
```

But it looks like even though there are Hebrew and a Farsi translations, neither are complete, and GTK+ falls back to English, in RTL mode.

(And after a short test, it looks like it's just missing translations)

**Comment by medicamin** [comment]
> (And after a short test, it looks like it's just missing translations)
I translated some statements to Farsi, placed the file to:
> /usr/share/locale/fa/LC_MESSAGES/FeedReader.mo 
and started the app with:
> LC_ALL=fa_IR.UTF-8 feedreader
I could see the translations are applied into the interface, but everything is in LTR. so it's not just missing translations.

**Comment by yotamN** [comment]
I think the first step should be to support RTL in the article view, with atom format you can check the `xml:lang` attribute and with RSS 2 you can check the language element but I'm not sure where in the code the feed xml is being parsed.

**Comment by bilelmoussaoui** [comment]
The app itself is already working great with RTL
![screenshot from 2019-02-10 14-33-03](https://user-images.githubusercontent.com/7660997/52534350-d4089580-2d40-11e9-9cb2-6d35e12abd04.png)
The only remaining thing is to detect the language of the article and switch the direction in the ArticleView too.

**Comment by Monirzadeh** [comment]
@bilelmoussaoui  any progress?
is  it possible to make RTL for content and not mirror UI?
